### PR TITLE
1417 status response format

### LINKED ISSUE #1417: change /status response format
Maybe we should change Status to consist of NodeInfo, SyncInfo, ValidatorInfo.

SyncInfo contains all the LatestXxx and the Syncing bool. ValidatorInfo contains the pubkey and the voting power.

### PR DESCRIPTION
**Pull request by Bric3d**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG.md

Fixes #1417

### CODE REVIEW

File: rpc/core/types/responses.go [hunk 1]
```diff
@@ -54,19 +54,23 @@ func NewResultCommit(header *types.Header, commit *types.Commit,
...
-	PubKey            crypto.PubKey   `json:"pub_key"`
-	LatestBlockHash   cmn.HexBytes    `json:"latest_block_hash"`
-	LatestAppHash     cmn.HexBytes    `json:"latest_app_hash"`
-	LatestBlockHeight int64           `json:"latest_block_height"`
-	LatestBlockTime   time.Time       `json:"latest_block_time"`
-	Syncing           bool            `json:"syncing"`
-	ValidatorStatus   ValidatorStatus `json:"validator_status,omitempty"`
+	NodeInfo      p2p.NodeInfo  `json:"node_info"`
+	SyncInfo      SyncInfo      `json:"sync_info"`
+	ValidatorInfo ValidatorInfo `json:"validator_info,omitempty"`
```
Comment (melekes) [inline rpc/core/types/responses.go]: remove `omitempty`. pubkey is always present

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ BUG FIXES:
 - Graceful handling/recovery for apps that have non-determinism or fail to halt
 - Graceful handling/recovery for violations of safety, or liveness
 
+## Develop
```
Comment (melekes) [inline CHANGELOG.md]: this needs to be moved to `0.18.0`

**Comment by melekes** [comment]
🥇 🏎 🍰

**Comment by Bric3d** [comment]
Fixed both the changelox and the omit .

**Comment by melekes** [comment]
```
++ date +%T
+ echo '* [07:31:24] running p2p tests on a local docker network'
* [07:31:24] running p2p tests on a local docker network
+ bash /home/circleci/project/test/circleci/../p2p/test.sh tester
Error response from daemon: No such container: local_testnet_1
Error response from daemon: No such container: local_testnet_1
Error response from daemon: No such container: local_testnet_2
Error response from daemon: No such container: local_testnet_2
Error response from daemon: No such container: local_testnet_3
Error response from daemon: No such container: local_testnet_3
Error response from daemon: No such container: local_testnet_4
Error response from daemon: No such container: local_testnet_4
Error response from daemon: No such network: local_testnet
PersistentPeers: f3a437ae47bd01b14f29e9199138d104554d1b59@172.57.0.101:46656,fbc1209802dd5c8e0f05618efd18a581f78291b9@172.57.0.102:46656,f47c979740323445ce90a355b41d6f9f0d1b3347@172.57.0.103:46656,2529fc79222fccb77c58ffb54c0bf450f869be64@172.57.0.104:46656
bb65a6f4fb55e725e75c7b3b3f769732deea6115312256f4ee8c7a43c9aeffff
starting tendermint peer ID=1
bc7075553c5ebcefb45859917753963ab62336bd92390a193d840bc6424a84ef
starting tendermint peer ID=2
e80bc129258df3d968a4ec980b9c5c3ff4e095ece56a94366deddd5c97e91172
starting tendermint peer ID=3
2c5d92a9764c307cb8bc716817a897ab4a219cd61f13c51366e00b985516d164
starting tendermint peer ID=4
7c03abe8433407885ae1194ba7f4cb9f33c29021e39381a275de9e1192c70d53
starting test client container with CMD=test/p2p/basic/test.sh 4
Waiting for nodes to come online

... node 1 is up

... node 2 is up

... node 3 is up

... node 4 is up



Waiting for node 1 to connect to all peers ...

Waiting for node 1 to connect to all peers ...

Waiting for node 1 to connect to all peers ...

test/p2p/basic/test.sh: line 59: [: null: integer expression expected
```

you'll have to fix p2p tests, sorry

**Comment by Bric3d** [comment]
I can't find anything that would cause that test to fail to be honest, when I test it locally it also ends with a timeout on the develop branch.

**Comment by melekes** [comment]
`curl -s $addr/status | jq .result.latest_block_height`

needs to be changed to

`curl -s $addr/status | jq .result.sync_info.latest_block_height`

**Comment by melekes** [comment]
Also, could you scan the docs/ directory for the same thing?

**Comment by Bric3d** [comment]
Yes I missed that the json was used in the tests, I'll check out if it's used in other places.

**Comment by melekes** [comment]
Could you please rebase this against latest `develop`?

**Comment by Bric3d** [comment]
Do you really want me to put it in the changelog of the 0.18.0 ? It's marked as released for the 6 of April already.

**Comment by melekes** [comment]
0.19.0

**Comment by Bric3d** [comment]
Should be rebased properly, data race ruined the test_cover though.

**Comment by melekes** [comment]
Could you please rebase against latest develop? I fixed the data race (`test_cover`).

**Comment by Bric3d** [comment]
Should be alright now.

**Pull request merged**